internal/locate/region_request.go

```diff
@@ -573,18 +573,36 @@ func (state *accessFollower) next(bo *retry.Backoffer, selector *replicaSelector
 	if state.option.preferLeader {
 		state.lastIdx = state.leaderIdx
 	}
+	var offset int
```

Comment:
Peer selection can become quite complex when the `accessFollower` is used for all follower read and stale read processing, with various configurations. To prevent any potential issues, it may be beneficial to separate them into different categories.

`ReplicaSelector`  itself is quite complex too..

Reply:
After discussing with @you06, we could do some refactoring on the master branch like
```
match access_type {
     leade_only =>
     follower_read_random =>
     follower_read_replica_only =>
     follower_read_closet => 
     stale_read_random => 
     stale_read_closest =>
}
```
while keeping the change on release-6.5 as simple as possible to reduce the risk.

Reply:
Maybe the refactor can be in another PR?

Reply:
Ok let's merge this first